server/base/src/main/java/org/apache/accumulo/server/util/MetadataTableUtil.java

```diff
@@ -489,9 +486,10 @@ public static void deleteTable(TableId tableId, boolean insertDeletes, ServerCon
 
   public static void removeUnusedWALEntries(ServerContext context, KeyExtent extent,
       final List<LogEntry> entries, ZooLock zooLock) {
-    Ample.TabletMutator tabletMutator = context.getAmple().mutateTablet(extent);
-    entries.forEach(tabletMutator::deleteWal);
-    tabletMutator.putZooLock(zooLock);
+    TabletMutator tablet = context.getAmple().mutateTablet(extent);
+    entries.forEach(tablet::deleteWal);
+    tablet.putZooLock(zooLock);
+    tablet.mutate();
```

Comment:
Nice catch.  Seems like this was a bug in the existing code.

Reply:
This fixed #1217 :+1: :smiley_cat: :tada: